File: src/test/java/pe/codember/app/Challenge02Test.java
package pe.codember.app;

import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

@DisplayName("Testing challenge 2 from codember 2023, its a mini compiler!!")
class Challenge02Test {

  /*
  "#" Increases the numeric value by 1.
  "@" Decreases the numeric value by 1.
  "*" Multiplies the numeric value by itself.
  "&" Prints the current numeric value.
  */

  @Test
  @DisplayName("Should be returns a string depending on how many prints string (&) there are")
  void test() {
    assertEquals(
        "4",
        Challenge02.something("##*&"),
        "Ups fix your solution"
    );
  }

  @Test
  @DisplayName("Should be returns a string depending on how many prints string (&) there are")
  void test02() {
    assertEquals(
        "0243",
        Challenge02.something("&##&*&@&"),
        "Ups fix your solution"
    );
  }

/*  @Test
  @DisplayName("This test is the solution to send")
  void finalTest() {
    assertEquals(
        "024899455",
        Challenge02.something(Challenge02.input),
        "Ups fix your solution"
    );
  }*/

}

File: src/test/typescript/challenge02.test.ts
import { compileThisWord } from '../../main/typescript/challenge02'

/*
"#" Increases the numeric value by 1.
"@" Decreases the numeric value by 1.
"*" Multiplies the numeric value by itself.
"&" Prints the current numeric value.
*/

describe('Testing challenge 2 from codember 2023, its a mini compiler!!', () => {
  test('Should be returns a string depending on how many prints string (&) there are', () => {
    const input = '##*&'
    const shouldBe = '4'

    expect(compileThisWord(input)).toBe<string>(shouldBe)
  })

  test('Should be returns a string depending on how many prints string (&) there are', () => {
    const input = '&##&*&@&'
    const shouldBe = '0243'
    expect(compileThisWord(input)).toBe<string>(shouldBe)
  })
})
